src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```diff
@@ -0,0 +1,254 @@
+    }
+    public static boolean isSupportedOldRefapp(Artifact artifact){
+        return isSupportedOldRefapp(artifact.getArtifactId(), artifact.getVersion());
+    }
+
+    public DistroProperties downloadDistroProperties(File path, Artifact artifact) throws MojoExecutionException {
+        artifact.setDestFileName("openmrs-distro.jar");
+        List<MojoExecutor.Element> artifactItems = new ArrayList<MojoExecutor.Element>();
+        MojoExecutor.Element element = artifact.toElement(path.toString());
+        artifactItems.add(element);
+
+        executeMojo(
+                plugin(
+                        groupId(SDKConstants.PLUGIN_DEPENDENCIES_GROUP_ID),
+                        artifactId(SDKConstants.PLUGIN_DEPENDENCIES_ARTIFACT_ID),
+                        version(SDKConstants.PLUGIN_DEPENDENCIES_VERSION)
+                ),
+                goal("copy"),
+                configuration(
+                        element(name("artifactItems"), artifactItems.toArray(new Element[0]))
+                ),
+                executionEnvironment(mavenProject, mavenSession, pluginManager)
+        );
+
+        DistroProperties distroProperties = null;
+        File file = new File(path, artifact.getDestFileName());
+        ZipFile zipFile = null;
+        try {
+            zipFile = new ZipFile(file);
+
+            Enumeration<? extends ZipEntry> entries = zipFile.entries();
+
+            while(entries.hasMoreElements()){
+                ZipEntry zipEntry = entries.nextElement();
+                if("openmrs-distro.properties".equals(zipEntry.getName())){
+                    Properties properties = new Properties();
+                    properties.load(zipFile.getInputStream(zipEntry));
+                    distroProperties = new DistroProperties(properties);
+                }
+            }
+
+            zipFile.close();
+
+
+        } catch (IOException e) {
+            throw new RuntimeException("Could not read " + file.toString(), e);
+        } finally {
+            IOUtils.closeQuietly(zipFile);
+            file.delete();
+        }
+
+        return distroProperties;
+    }
+
+    public DistroProperties downloadDistroProperties(File serverPath, Server server) throws MojoExecutionException {
+        Artifact artifact = new Artifact(server.getDistroArtifactId(), server.getVersion(), server.getDistroGroupId(), "jar");
+        return downloadDistroProperties(serverPath, artifact);
+    }
+
+    /**
+     * Distro can be passed in two ways: either as maven artifact identifier or path to distro file
+     * Returns null if string is invalid as path or identifier
+     * @param distro
+     * @return
+     */
+    public DistroProperties retrieveDistroProperties(String distro) throws MojoExecutionException {
+        DistroProperties result;
+        result = getDistroPropertiesFromDir(new File(distro));
+        if(result==null){
+            Artifact artifact = parseDistroArtifact(distro);
+            if(artifact!=null){
```

Comment:
Eliminate "if", parseDistroArtifact should throw an exception in such a case and end the execution.